mmpose/models/necks/posewarper_neck.py

```diff
@@ -0,0 +1,311 @@
+import mmcv
+import torch
+import torch.nn as nn
+from mmcv.cnn import (build_conv_layer, build_norm_layer, constant_init,
+                      normal_init)
+from mmcv.ops import DeformConv2d
+from torch.nn.modules.batchnorm import _BatchNorm
+
+from mmpose.models.utils.ops import resize
+from ..backbones.resnet import BasicBlock, Bottleneck
+from ..builder import NECKS
+
+
+@NECKS.register_module()
+class PoseWarperNeck(nn.Module):
+    """PoseWarper neck. Paper ref: Bertasius, Gedas, et al. "Learning temporal
+    pose estimation from sparsely-labeled videos." arXiv preprint
+    arXiv:1906.04016 (2019).
+
+    <`https://arxiv.org/abs/1906.04016`>
+
+    Args:
+        in_channels (int): Number of input channels from backbone
+        out_channels (int): Number of output channels
+        inner_channels (int): Number of intermediate channels
+        deform_groups (int): Number of groups in the deformable conv
+        dilations (list|tuple): different dilations of the offset conv layers
+        extra (dict): config of the conv layer to get heatmap
+        res_blocks (dict): config of residual blocks
+        offsets (dict): config of offset conv layer
+        deform_conv (dict): config of defomrable conv layer
+        in_index (int|Sequence[int]): Input feature index. Default: 0
+        input_transform (str|None): Transformation type of input features.
+            Options: 'resize_concat', 'multiple_select', None.
+            'resize_concat': Multiple feature maps will be resize to the
+                same size as first one and than concat together.
+                Usually used in FCN head of HRNet.
+            'multiple_select': Multiple feature maps will be bundle into
+                a list and passed into decode head.
+            None: Only one select feature map is allowed.
+            Default: None.
+        freeze_trans_layer (bool): Whether to freeze the transition layer
+            (stop grad and set eval mode). Default: False.
+        norm_eval (bool): Whether to set norm layers to eval mode, namely,
+            freeze running stats (mean and var). Note: Effect on Batch Norm
+            and its variants only. Default: False
+        im2col_step: the argument `im2col_step` in deform_conv
+        least_mmcv_version (str): should use version newer than this version
+    """
+    blocks_dict = {'BASIC': BasicBlock, 'BOTTLENECK': Bottleneck}
+
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 inner_channels,
+                 deform_groups=17,
+                 dilations=(3, 6, 12, 18, 24),
+                 extra=None,
+                 res_blocks={},
+                 offsets={},
+                 deform_conv={},
+                 in_index=0,
+                 input_transform=None,
+                 freeze_trans_layer=False,
+                 norm_eval=False,
+                 im2col_step=80,
+                 least_mmcv_version='1.3.16'):
```

Comment:
`least_mmcv_version` should not be an argument. Maybe make it a class attribute, or just hard code it in the version check. And I suggest using the name `minimum_mmcv_version`.

Reply:
OK, I will make it a class attribute and use the name `minimum_mmcv_version` instead.